docs/cli/token.md

```diff
@@ -161,3 +161,10 @@ If you do not specify a new token, one will be generated for you.
 `--token` value      | Existing token used to join a server or agent to a cluster [$K3S_TOKEN]                   
 `--new-token` value  | New token that replaces existing token   
 
+:::warning
+Data encrypted before the rotation will need the old server token to decrypt
```

Comment:
We attempt to migrate all data that depends on those keys with https://github.com/k3s-io/k3s/blob/master/pkg/cluster/storage.go#L304. What data is not being reencrypted?

Reply:
Yeah I don't think this is accurate. Changing the server token does NOT change the encryption config, so the same key is used to encrypt/decrypt the secrets.

Reply:
Ok, I got confused by a comment. The only encrypted object which will need the old server token are old snapshots then, right? I wanted to add this warning because when executing this command, you get a warning with something like `keep the old token` but perhaps it is too late once you read that warning? The token gets replaced a few seconds after that

Reply:
The snapshots themselves are not encrypted either. Any snapshots taken with the old token value, will still have the bootstrap data IN the snapshot encrypted with the old token, so you'd need to use the old token when restoring it.

Reply:
Thanks Brad, for some reason I was misunderstanding this. I thought the snapshot was also encrypted :facepalm:

Reply:
Its kinda complicated. Everything in this struct:
https://github.com/k3s-io/k3s/blob/cd4ddedbc9782cbe9b5dcc411df2addae7b2f3b4/pkg/daemons/config/types.go#L291-L307

Is read from disk is stuffed into a map, where the key is the field name and the value is the content of file at that path on disk along with a modification timestamp. This is then serialized to JSON and encrypted with the key.

You can find the code for translating from `ControlRuntimeBootstrap` to `map[string]File` here:
https://github.com/k3s-io/k3s/blob/cd4ddedbc9782cbe9b5dcc411df2addae7b2f3b4/pkg/bootstrap/bootstrap.go#L23-L54

Reply:
Thanks Brad. Since the struct https://github.com/k3s-io/k3s/blob/cd4ddedbc9782cbe9b5dcc411df2addae7b2f3b4/pkg/daemons/config/types.go#L291-L307 might keep changing in the future, I think it is better if we remove what comes after `such as` to avoid it getting outdated again. I rewrote the paragraph a bit